src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import { OpentableIcon as icon } from '@godaddy-wordpress/coblocks-icons';
+import { useState, useEffect } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import Controls from './controls';
+import OpenTable from './opentable';
+import InspectorControls from './inspector';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+import {
+	Placeholder,
+	Button,
+	withNotices,
+	Icon,
+	FormTokenField,
+} from '@wordpress/components';
+
+const Edit = ( props ) => {
+	// const onSelectImages = ( images ) => {
+	// 	props.setAttributes( { images: images.map( ( image ) => pickRelevantMediaFiles( image ) ) } );
+	// };
+
+	// const onDropImages = ( images ) => {
+	// 	const imagesNormalized = images.map( ( image ) => pickRelevantMediaFiles( image ) );
+	// 	const currentImages = props.attributes.images || [];
+	// 	props.setAttributes( { images: currentImages.concat( imagesNormalized )	} );
+	// };
+
+	// const { className, noticeOperations, attributes, noticeUI, isSelected } = props;
+
+	const [ ridField, setRidField ] = useState( [] );
+
+	const [ queryResults, setQueryResults ] = useState( [] );
+	const { className, attributes } = props;
+
+	//searches the opentable reservation network for restaurants with the given name or ID
+	const searchRestaurants = ( token ) => {
+		fetch(
+			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=15' +
+				'&query=' +
+				encodeURIComponent( token )
+		)
+			.then( ( response ) => response.json() )
+			.then( ( json ) => {
+				setQueryResults( [] );
+				for ( const item in json.items ) {
+					const name = decodeURIComponent( json.items[ item ].name ) + ' (' + json.items[ item ].rid + ')';
+					setQueryResults( ( oldQueryResults ) => [ ...oldQueryResults, name ] );
+				}
+			} );
```

Comment:
On line 56 `setQueryResults` setter function is used in a loop. While we preserve old results, there would be excess calls to this function where it could be avoided. I would suggest setting a variable on line 53, add to it on line 56, and finally execute `setQueryResults` outside of the loop on line 58.